rct_optimizations/test/pnp_utest.cpp

```diff
@@ -0,0 +1,43 @@
+#include <rct_optimizations/experimental/pnp.h>>
+#include <gtest/gtest.h>
+
+using namespace rct_optimizations;
+
+TEST(PnPTest, 3DValidation)
+{
+  //correspondences: do easy offset between target & camera (ie make set translate 0.5 +x)
+  //setup params
+  PnPProblem3D setup;
+
+  //testing with only one observation
+  Correspondence3D3D::Set corrs;
+  corrs.reserve(9);
+
+  Eigen::Vector3d origin(0.0,0.0,0.0);
+
+  for(std::size_t i=0; i < 3; ++i)
+  {
+    for (std::size_t j =0; j < 3; j++)
+    {
+      Correspondence3D3D c;
+      Eigen::Vector3d world_point =Eigen::Vector3d(double(i), double(j), double((i*j)%3));
+      Eigen::Vector3d camera_point = world_point;
+      c.in_target = world_point;
+      c.in_image = camera_point;
+
+      corrs.push_back(c);
+    }
+  }
+
+  setup.correspondences = corrs;
+  setup.camera_to_target_guess = Eigen::Isometry3d::Identity();
```

Comment:
1. Please use the correspondence generation functions we've already created to make this unit test consistent with the others and to reduce code maintenance
1. Please put the camera in a realistic position for viewing the targets. These tests should be relatively representative of real use-cases (for understandability of the tests, at the very least). I would suggest putting the camera directly over the target by ~1.0 m and rotating it such that the Z-axis of the camera points at the target